src/feature_annotation/score_genes_scanpy/script.py

```diff
@@ -0,0 +1,89 @@
+import scanpy as sc
+import anndata as ad
+import pandas as pd
+import mudata as mu
+import sys
+
+## VIASH START
+par = {
+    "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_mms.h5mu",
+    "modality": "rna",
+    "input_layer": "log_normalized",
+    "gene_list_file": None,
+    "gene_list": ["UBE2C", "BIRC5", "TPX2"],
+    "gene_pool_file": None,
+    "gene_pool": None,
+    "var_gene_names": "gene_symbol",
+    "output": "output.h5mu",
+    "ctrl_size": 50,
+    "n_bins": 25,
+    "obs_score": "score",
+    "random_state": 0,
+    "output_compression": "gzip",
+    "allow_missing_genes": False,
+}
+meta = {
+    "resources_dir": "src/feature_annotation/score_genes_scanpy"
+}
+## VIASH END
+
+sys.path.append(meta["resources_dir"])
+
+from helper import read_gene_list
+
+# read data
+mdata = mu.read(par["input"])
+input_adata = mdata.mod[par["modality"]]
+
+if par["var_gene_names"]:
+    gene_names = list(input_adata.var[par["var_gene_names"]])
+else:
+    gene_names = list(input_adata.var.index)
+
+# check if var index is unique
+if not input_adata.var.index.is_unique:
+    raise ValueError("var index is not unique")
+
+
+# read gene list
+gene_list = read_gene_list(
+    par,
+    gene_names,
+    "gene_list",
+    "gene_list_file"
+)
+gene_pool = read_gene_list(
+    par,
+    gene_names,
+    "gene_pool",
+    "gene_pool_file",
+    required=False
+)
+
+# find matching index names for given genes
+gene_list_index = input_adata.var.index[[gene in gene_list for gene in gene_names]]
```

Comment:
I'm not sure, but I think this also works and uses the intended pandas methods. Could you test it?
```suggestion
gene_pool_index = input_adata.var_names.intersection(gene_names.reindex(gene_list)[0])
```

Reply:
My previous comment will _not_ work when gene_names contains duplicate entries

What I would do is change starting from line 38:

```python3
gene_names_index = input_adata.var[par["var_gene_names"]] if par["var_gene_names"] else input_adata.var_names
gene_names = pd.Series(input_adata.var_names, index=gene_names_index)
```

Make sure that `read_gene_list` works with the Series object _or_ just pass `gene_names.index` as the second argument

And then use: 
```python3
gene_list_index = gene_names.loc[gene_list].tolist()
```
But perhaps tbd with @rcannood

Reply:
@rcannood i commited those changes in [e87c9bd](https://github.com/openpipelines-bio/openpipeline/pull/703#:~:text=update%20index%20matching-,e87c9bd,-DriesSchaumont%20requested%20changes,), so can be merged/reverted back